web/src/js/ducks/modes.ts

```diff
@@ -0,0 +1,32 @@
+import { combineReducers } from "redux";
+import regularReducer from "./modes/regularDuck";
+import { fetchApi } from "../utils";
+
+export interface ModeState {
+    active: boolean;
+    listen_port?: number;
+    listen_host?: string;
+}
+
+export const updateMode = async (mode, active) => {
```

Comment:
When calling `await fetchApi.put("/options", { mode: ... })` we always need to pass all currently active modes. So I would propose that updateMode does not take any arguments, but goes and interrogates the current state to assemble the list of modes and the call `fetchApi`. 

`toggleRegular` would just dispatch an action to update `active` and then call `updateMode` (without arguments).

Reply:
okay:)